src/jit/simdcodegenxarch.cpp

```diff
@@ -951,6 +1088,716 @@ void CodeGen::genSIMDIntrinsicUnOp(GenTreeSIMD* simdNode)
+                                    var_types       baseType,
+                                    regNumber       tmpReg,
+                                    regNumber       tmpIntReg,
+                                    regNumber       targetReg)
+{
+    instruction ins = getOpForSIMDIntrinsic(intrinsicID, baseType);
+    if (intrinsicID == SIMDIntrinsicConvertToDouble)
+    {
+        // Note that for mov_xmm2i, the int register is always in the reg2 position
+        inst_RV_RV(INS_mov_xmm2i, tmpReg, tmpIntReg, TYP_LONG);
+        inst_RV_RV(ins, targetReg, tmpIntReg, baseType, emitActualTypeSize(baseType));
+    }
+    else
+    {
+        inst_RV_RV(ins, tmpIntReg, tmpReg, baseType, emitActualTypeSize(baseType));
+        inst_RV_RV(INS_mov_i2xmm, targetReg, tmpIntReg, TYP_LONG);
+    }
+}
+
+//----------------------------------------------------------------------------------
+// genSIMDIntrinsic64BitConvert: Generate code for 64-bit SIMD Convert (long/ulong <-> double)
+//
+// Arguments:
+//    simdNode - The GT_SIMD node
+//
+// Notes:
+//    There are no instructions for converting to/from 64-bit integers, so for these we
+//    do the conversion an element at a time.
+//
+void CodeGen::genSIMDIntrinsic64BitConvert(GenTreeSIMD* simdNode)
+{
+    SIMDIntrinsicID intrinsicID = simdNode->gtSIMDIntrinsicID;
+    assert((intrinsicID == SIMDIntrinsicConvertToDouble) || (intrinsicID == SIMDIntrinsicConvertToInt64) ||
+           (intrinsicID == SIMDIntrinsicConvertToUInt64));
+
+    GenTree*  op1       = simdNode->gtGetOp1();
+    var_types baseType  = simdNode->gtSIMDBaseType;
+    regNumber targetReg = simdNode->gtRegNum;
+    assert(targetReg != REG_NA);
+    var_types      simdType  = simdNode->TypeGet();
+    regNumber      op1Reg    = genConsumeReg(op1);
+    regNumber      tmpIntReg = genRegNumFromMask(simdNode->gtRsvdRegs & RBM_ALLINT);
+    regNumber      tmpReg;
+    regNumber      tmpReg2;
+    regNumber      tmpReg3;
+    InstructionSet iset = compiler->getSIMDInstructionSet();
+
+#ifdef _TARGET_AMD64_
+    if (iset == InstructionSet_AVX || (baseType == TYP_ULONG))
+#else
+    if (baseType == TYP_LONG)
+    {
+        regMaskTP tmpRegsMask = (simdNode->gtRsvdRegs & RBM_ALLFLOAT);
+        assert(tmpRegsMask != RBM_NONE);
+        assert(genCountBits(tmpRegsMask) == 3);
+
+        regMaskTP tmpReg1Mask = genFindLowestBit(tmpRegsMask);
+        tmpRegsMask           = tmpRegsMask - tmpReg1Mask;
+        regMaskTP tmpReg2Mask = genFindLowestBit(tmpRegsMask);
+        tmpRegsMask           = tmpRegsMask - tmpReg2Mask;
+        regMaskTP tmpReg3Mask = genFindLowestBit(tmpRegsMask);
+
+        tmpReg  = genRegNumFromMask(tmpReg1Mask);
+        tmpReg2 = genRegNumFromMask(tmpReg2Mask);
+        tmpReg3 = genRegNumFromMask(tmpReg3Mask);
+        assert(tmpReg != op1Reg && tmpReg2 != op1Reg && tmpReg3 != op1Reg);
+    }
+    else if (iset == InstructionSet_AVX || (baseType == TYP_ULONG))
+#endif
+    {
```

Comment:
This could be:
```
#ifdef _TARGET_X86_
    if (baseType == TYP_LONG)
    {
    ...
    }
#else
    else if (iset == InstructionSet_AVX || (baseType == TYP_ULONG))
```